### DISCUSSION TITLE
Question: Will uninstalling the app persist the storage?

### DISCUSSION THREAD

**Discussion by rahgurung** [react-native-async-storage/async-storage#835] [Q&A] [answered]
A little explanation on why and how would help too.

**Comment by tido64** [comment, marked as answer]
Short answer: AsyncStorage does not control this and it depends on your app and/or the platform.

Longer answer: On Android, it seems that the default behaviour is to delete only the app and keep the data around. This means that if you re-install the app, you'll get the state prior to removing the app. Apparently, you can disable this by setting `android:allowBackup="false"`. On iOS, it seems that deleting the app also removes the associated data. You can opt into iCloud, which should backup the data and restore it on re-install.

**Comment by rahgurung** [comment]
If user updates the app, it will be equivalent of a reinstall right? Sorry if this is a irrelavant question.

**Reply by tido64** [reply]
When the user updates the app, the app is just replaced. The data is not touched.

**Reply by hourzn** [reply]
> When the user updates the app, the app is just replaced. The data is not touched.

Hi Tido,

My app data was cleared upon updating in the iOS App Store. Is there anyway to retrieve my previously stored data using AsyncStorage?
